sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -1221,17 +1279,27 @@ private static class BigtableWriterFn
 
     private final BigtableServiceFactory factory;
     private final BigtableServiceFactory.ConfigId id;
+    private final Coder<KV<ByteString, Iterable<Mutation>>> inputCoder;
+    private final BadRecordRouter badRecordRouter;
+
+    private transient Set<KV<BigtableWriteException, BoundedWindow>> badRecords = new HashSet<>();
+
+    private transient Set<CompletionStage<MutateRowResponse>> unbatchedWrites = new HashSet<>();
```

Comment:
These instances will never to be used (startBundle will always overwrite and when its deserialized, it will be null). I think it would be nicer to initialize them as null to begin with

Reply:
sg